### PR TITLE
Transition from Travis CI to github workflow

### PR DESCRIPTION
**Pull request by franchuterivera**
- Added 4 cores to our run
- Added dist check from openml
- Updated other openml github checks to our use case

### CODE REVIEW

File: .github/workflows/pre-commit.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: pre-commit
+
+on: [push]
```
Comment (mfeurer) [inline .github/workflows/pre-commit.yaml:3]: Shouldn't there be a `pull_request`, too?

File: .travis.yml [hunk 2]
```diff
@@ -1,83 +0,0 @@
...
-  - MODULE=autosklearn
-  - MINICONDA_URL="https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh"
-
-matrix:
-  allow_failures:
-  - os: osx
-
-  include:
-  - os: linux
-    env: DISTRIB="conda" DOCPUSH="true" PYTHON="3.7" SKIP_TESTS="true"
```
Comment (mfeurer) [inline .travis.yml:24]: The new tests do no longer test whether we can install in a conda environment. Could you please re-add those tests?

File: .travis.yml [hunk 3]
```diff
@@ -1,83 +0,0 @@
...
-git:
-  depth: 5
-
-before_install:
-  - source ci_scripts/install_env.sh
-
-install:
-  - source ci_scripts/install.sh
-
-script: bash ci_scripts/test.sh
```
Comment (mfeurer) [inline .travis.yml:71]: Some of these scripts can probably be deleted now?

File: .github/workflows/pytest.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+name: Tests
```
Comment (mfeurer) [inline .github/workflows/pytest.yml:1]: Do you set the environment variables regarding the number of threads to use as on travis-ci somewhere?

File: .github/workflows/pytest.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+        include:
+          - python-version: 3.8
+            code-cov: true
+      fail-fast:  false
+      max-parallel: 2
```
Comment (mfeurer) [inline .github/workflows/pytest.yml:16]: There's no need to limit ourselves here.

File: setup.py [hunk 6]
```diff
@@ -32,9 +32,17 @@
         "pytest-xdist",
         "pytest-timeout",
         "flaky",
-        "pytest-cov",
         "openml",
-    ]
+        "pre-commit",
```
Comment (mfeurer) [inline setup.py:36]: I really like this, makes the whole installation process way more organized!

File: .github/workflows/dist.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+name: dist-check
```
Comment (mfeurer) [inline .github/workflows/dist.yml:1]: What worries me about the dist check is that we don't run auto-sklearn afterwards. This is mostly to check that we don't forget to include anything that is vital to run in the distribution. Can you maybe factor that into the actual tests (similar to  codecov)

File: .github/workflows/pytest.yml [hunk 8]
```diff
@@ -9,26 +9,65 @@ jobs:
...
+      if: matrix.use-conda == false && matrix.use-dist == true
+      run: |
+        python -m pip install --upgrade pip
+        sudo apt-get update
+        sudo apt-get remove swig
+        sudo apt-get install swig3.0
+        sudo ln -s /usr/bin/swig3.0 /usr/bin/swig
+        python setup.py sdist
+        last_dist=$(ls -t dist/auto-sklearn-*.tar.gz | head -n 1)
+        pip install $last_dist
```
Comment (mfeurer) [inline .github/workflows/pytest.yml:54]: This is now missing:
```
    twine_output=`twine check "$dist"`
    if [[ "$twine_output" != "Checking $dist: PASSED" ]]; then
        echo $twine_output
        exit 1
    else
        echo "Check with Twine: OK: $twine_output"
    fi
```

File: .github/workflows/pytest.yml [hunk 9]
```diff
@@ -9,26 +9,65 @@ jobs:
...
+        $CONDA/envs/testenv/bin/python3 -m pytest --durations=20 -sv $codecov test
     - name: Run tests
+      if: matrix.use-conda == false
       run: |
+        export OPENBLAS_NUM_THREADS=1
+        export OMP_NUM_THREADS=1
+        export MKL_NUM_THREADS=1
         if [ ${{ matrix.code-cov }} ]; then codecov='--cov=autosklearn --cov-report=xml'; fi
-        pytest -sv $codecov test
+        pytest --durations=20 -sv $codecov test
```
Comment (mfeurer) [inline .github/workflows/pytest.yml:70]: Why do you need to pass `durations=20`?

File: .github/workflows/pytest.yml (hunk 8 above)
Comment (franchuterivera) [reply .github/workflows/pytest.yml:58]: Hi Matthias, please let me know if this is not the check you were looking for:
https://github.com/automl/auto-sklearn/blob/40e59eb0c4105ad75c17c20cac8e54dd27a935d7/.github/workflows/dist.yml#L21

**Comment by franchuterivera** [comment]
Only Docs failing (plus random error on delete non candidate models).

Previous Docs passed ok.

**Review by mfeurer — APPROVED** [review]

File: .github/workflows/pytest.yml (hunk 8 above)
Comment (mfeurer) [reply .github/workflows/pytest.yml:58]: Yes, it is, but it does net check that the output value is fine.

**Review by mfeurer — CHANGES_REQUESTED** [review]

**Review by mfeurer — APPROVED** [review]

**Pull request merged**